### ISSUE TITLE
Windows Path Destroyed

### ISSUE THREAD

**Issue by DACN** [shoes/shoes3#421]
I have just installed shoes (shoes-3.3.6-gtk3-32.exe) on windows 10. In doing so, it has just destroyed the Path environment variable, leaving shoes at the only entry. The installation of a tonne of other programs destroyed by installing shoes.

**Comment by ccoupe** [comment]
Damn! That problem was fixed a few years ago. It must have come back. So Sorry! 

FWIW, it's likely to be the NSIS installer was built w/o 'long string' support.

**Comment by DACN** [comment]
Yes - for some reason the 'default versions' of NSIS which you download do
not have long string support. You need to look slightly more carefully to
get one with long string support. So, agree - that is a very plausible
cause.

D

**Comment by ccoupe** [comment]
FWIW, I've tested Shoes 3.3.7 (beta) and it cut off a path that was _roughly_ 2700 long. That's with the long string patch (supposedly).  How big was your path, @DACN ?

**Comment by ccoupe** [comment]
[Interesting backaground](https://superuser.com/questions/1070272/why-does-windows-).  If I understand it, 2048 characters is the _practical_ limit, and you'd want to be a bit less than that.

**Comment by ccoupe** [comment]
Fixed. If the PATH + addition > NSIS_MAX_STRLEN then user will get a warning and the path won't be modified. That works even if the big string patch is not used.

**Comment by DACN** [comment]
Thanks

David

**Comment by Ryokai96** [comment]
I have just installed  (shoes-3.3.7-gtk3-32.exe) on Windows7 and Windows10, it have the same problem, I haven't got warning and my windows path has modified. Please fix it.

**Comment by ccoupe** [comment]
@Ryokai96 - Sorry for the difficulty. It must be an off-by-one problem.  Be aware the Shoes is no longer being maintained so this problem is unlikely to be fixed.
